src/main/java/chatbot/command/DeadlineCommand.java

```diff
@@ -0,0 +1,49 @@
+package chatbot.command;
+
+import chatbot.task.Deadline;
+import chatbot.task.Task;
+import chatbot.task.TaskList;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Arrays;
+
+public class DeadlineCommand implements Command {
+    public static final String TRIGGER = "deadline";
+    public static final String FORMAT =
+            "Command Format: \"" + TRIGGER + " <desc> /by <date> <time>\"\n" + "Date format: YYYY-MM-DD\n"
+                    + "Time format: HH:MM or HH:MM:SS";
+
+    @Override
+    public CommandOutput execute(String[] input, TaskList taskList) {
+        // Parse input.
+        String desc = "";
+        LocalDate date = null;
+        LocalTime time = null;
+        try {
+            int i = 1;
+            while (i < input.length && !input[i].equals("/by")) {
+                ++i;
+            }
+            desc = String.join(" ", Arrays.asList(input).subList(1, i));
```

Comment:
Could look a bit less complicated-looking if you separate the `Arrays.asList(input).subList(1, i)` part to another variable.